### PR TITLE
Freetube: initial commit

### PR DESCRIPTION
**Pull request by Kreyren**
Fixes: https://github.com/nix-community/home-manager/issues/2152

### TODO
- [ ] Figure out how is this app storing it's data on NixOS
#### Before merge
- [ ] Add myself in the maintainers list (https://github.com/nix-community/home-manager/pull/2159#discussion_r664106592)
- [ ] Run the format script

### Description

First commit, have no idea what am i doing while making the code look somewhat sane and not knowing how to test it, help

### Checklist

- [ ] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [ ] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/freetube.nix
Comment (berbiche) [inline modules/programs/freetube.nix:13]: We don't support these platforms at this time so this can be removed.
```suggestion
```

File: modules/programs/freetube.nix
Comment (berbiche) [inline modules/programs/freetube.nix:25]: We don't have a machine to test this in the CI and AFAIK we do not currently support this platform.
```suggestion
```

File: modules/programs/freetube.nix
Comment (berbiche) [inline modules/programs/freetube.nix:52]: Make sure to run the format script per the documentation.

File: modules/programs/freetube.nix
Comment (berbiche) [inline modules/programs/freetube.nix:32]: You need to add yourself to the maintainers list of HM if you are not a maintainer in Nixpkgs' `maintainer-list.nix`.

This is documented in `doc/contributing.adoc`.

``` console
$ nix eval --impure --expr 'with import <nixpkgs> {}; lib.maintainers ? kreyren'
false
```

**Review by berbiche — CHANGES_REQUESTED** [review]
Hi,

Left some comments to help you out.

What set of options can you configure for Freetube?
How does it store the user secrets for authentication (is the password stored in the configuration file)?

**Comment by Kreyren** [comment]
> What set of options can you configure for Freetube? @berbiche

The main motivation is being able to set subscriptions ideally in a way that can be transferred to other packages such as NewPipe as well.

Then having option to set up the proxy setting

> How does it store the user secrets for authentication (is the password stored in the configuration file)? @berbiche

Mentioned in the code.

On Linux it should be using `$HOME/.config/FreeTube`, but it seems that NixOS is handling that differently (researching atm)

See https://docs.freetubeapp.io/usage/data-location/

File: modules/programs/freetube.nix
Comment (Kreyren) [reply modules/programs/freetube.nix:13]: NixOS has logic for them so i think it could be integrated in case home-manager decides to support them?

(i like to make my code crossplatform whenever possible)

File: modules/programs/freetube.nix
Comment (Kreyren) [reply modules/programs/freetube.nix:25]: GitHubCI offers Windows Machine to run tests on, i can try to integrate if if you want (would do a separate MR)

File: modules/programs/freetube.nix
Comment (Kreyren) [reply modules/programs/freetube.nix:52]: will do once it's ready to merge

**Comment by berbiche** [comment]
> > How does it store the user secrets for authentication (is the password stored in the configuration file)? @berbiche
> 
> Mentioned in the code.
> 
> On Linux it should be using `$HOME/.config/FreeTube`, but it seems that NixOS is handling that differently (researching atm)
> 
> See https://docs.freetubeapp.io/usage/data-location/

@Kreyren I suppose the login secrets are stored in profiles.db then?

**Pull request closed without merging**
